src/pint/models/parameter.py

```diff
@@ -1581,19 +1593,20 @@ def select_toa_mask(self, toas):
         # get the table columns
         # TODO Right now it is only supports mjd, freq, tel, and flagkeys,
         # We need to consider some more complicated situation
+        # THis will have a problem if the key has '-' in it.
         key = self.key.replace("-", "")
         tbl = toas.table
         if (
-            key.lower() not in column_match.keys()
+            self.key.lower() not in column_match.keys()
```

Comment:
`in` on dictionaries does the same thing as on their keys; you can leave out the `.keys()` here.

Reply:
This is fixed in the update

Reply:
I'm not sure what you mean?